orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -132,10 +190,60 @@ private TaskResult handleError(
         monitorDefinition,
         e);
 
-    return TaskResult.builder(ExecutionStatus.SUCCEEDED)
-        // TODO(mvulfson)
-        // .context()
-        .build();
+    String userMessage =
+        String.format(
+            "Failed to get a valid response from deployment monitor %s, proceeding anyway because this deployment monitor is configured to not failOnError",
+            monitorDefinition.getName());
+
+    return buildTaskResult(TaskResult.builder(ExecutionStatus.SUCCEEDED), userMessage);
+  }
+
+  private void ensureMonitorDefinition(Stage stage) {
+    if (this.stage == null) {
```

Comment:
This should be `!= null` right?

```suggestion
    if (this.stage != null) {
```

Reply:
this just sets instance vars (sort of lazily) so `==` is right. basically this is called from a bunch of spots to "init" the class instance vars

Reply:
Doh! 🤦‍♂ Of course. Please forget that I asked that.